### ISSUE TITLE
Compilation errors "int32_t cannot be overloaded"

### ISSUE THREAD

**Issue by kke** [argandas/ardubson#1]
Getting errors while trying to compile:

```c++
In file included from .piolibdeps/ardubson/ardubson.h:15:0,
from src/main.cpp:17:
.piolibdeps/ardubson/ardubsonElement.h:32:14: error: 'void BSONElement::Value(int32_t)' 
cannot be overloaded
void Value(int32_t value);
^
.piolibdeps/ardubson/ardubsonElement.h:31:14: error: with 'void BSONElement::Value(int)'
void Value(int value);
^
In file included from .piolibdeps/ardubson/ardubson.h:16:0,
from src/main.cpp:17:
.piolibdeps/ardubson/ardubsonObjBuilder.h:42:25: error: 'BSONObjBuilder& BSONObjBuilder::append(const char*, int32_t)'
cannot be overloaded
BSONObjBuilder& append(const char *key, int32_t value);
^
.piolibdeps/ardubson/ardubsonObjBuilder.h:41:25: error: with 'BSONObjBuilder& BSONObjBuilder::append(const char*, int)'
BSONObjBuilder& append(const char *key, int value);
^
```

**Comment by guyikcgg** [comment]
I found the same problem. The issue is that `int` type in your system is actually the same that `int32_t`. This implies that you are not `overloading BSONElement::Value()` or `BSONObjBuilder::append()` but redefining it. To solve the issue, the use of `int` should be avoided, using `int16_t` standard type instead.

The issue is solved here: https://github.com/guyikcgg/ardubson/commit/9ecd2f6c9107832dd56859551514fe9d5243d370 I will submit some pull requests soon ;)

**Comment by argandas** [comment]
Hello all, this have been fixed in recent changes, please check v1.3.0
